metals/src/main/scala/scala/meta/internal/metals/DocumentHighlightProvider.scala

```diff
@@ -0,0 +1,41 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.{DocumentHighlight, TextDocumentPositionParams}
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.Semanticdbs
+
+class DocumentHighlightProvider(
+    definitionProvider: DefinitionProvider,
+    semanticdbs: Semanticdbs
+) {
+
+  def symbolOccurrences(
+      params: TextDocumentPositionParams
+  ): java.util.List[DocumentHighlight] = {
+    val source = params.getTextDocument.getUri.toAbsolutePath
+    val result = semanticdbs.textDocument(source)
+    result.documentIncludingStale
+      .flatMap { doc =>
+        val ResolvedSymbolOccurrence(editDistance, symbolOccurrence) =
+          definitionProvider.positionOccurrence(
+            source,
+            params,
+            doc,
+            includeLastCharacter = true
+          )
+        symbolOccurrence.map { occ =>
+          val allOccurrences = doc.occurrences
```

Comment:
For consistency with the style in the rest of the project, can we use for comprehensions instead?
```scala
    val highlights = for {
      doc <- result.documentIncludingStale.toList
      positionOccurrence = definitionProvider.positionOccurrence(
        source,
        params,
        doc,
        includeLastCharacter = true
      )
      occ <- positionOccurrence.occurrence.toList
      curr <- doc.occurrences
      if curr.symbol == occ.symbol
      range <- curr.range
      revised <- positionOccurrence.distance.toRevised(range.toLSP)
    } yield new DocumentHighlight(revised)
    highlights.asJava
```

It's fine to use `.toList` so we can mix options with lists.

Reply:
Changed